src/middleware/packages/core/service.js

```diff
@@ -13,32 +13,17 @@ const { WebAclService } = require('@semapps/webacl');
 const { WebfingerService } = require('@semapps/webfinger');
 const defaultOntologies = require('./config/ontologies.json');
 
+/**
+ * @typedef {import('http').ServerResponse} ServerResponse
+ * @typedef {import('http').IncomingMessage} IncomingMessage
+ * @typedef {import('moleculer').ServiceSchema} ServiceSchema
+ * @typedef {import('./service').CoreServiceSettings} CoreServiceSettings
+ */
+
+/** @type {import('moleculer').ServiceSchema<CoreServiceSettings>} */
 const CoreService = {
   name: 'core',
-  settings: {
-    baseUrl: null,
-    baseDir: null,
-    triplestore: {
-      url: null,
-      user: null,
-      password: null,
-      mainDataset: null
-    },
-    // Optional
-    containers: null,
-    jsonContext: null,
-    ontologies: null,
-    // Services configurations
-    activitypub: {},
-    api: {},
-    jsonld: {},
-    ldp: {},
-    signature: {},
-    sparqlEndpoint: {},
-    void: {},
-    webacl: {},
-    webfinger: {}
-  },
+  settings: {},
```

Comment:
I would remove the settings because, most of the time, they are used for default values. For the services configurations, we destructure the empty object and it will fail with an `undefined` value.